File: in_Java/0349 Intersection of Two Arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> hs = new HashSet<>();
        for(int i : nums1) hs.add(i);
        List<Integer> arr = new ArrayList<>();
        for(int i : nums2){
            if(hs.contains(i)){
                arr.add(i);
                hs.remove(i);
            }
        }
        int[] res = new int[arr.size()];
        for(int i = 0; i < arr.size(); i++) res[i] = arr.get(i);
        return res;
    }
}

File: in_Python/0349 Intersection of Two Arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        a = set()
        b = set()
        for i in nums1:
            a.add(i)
        for i in nums2:
            b.add(i)
        res = []
        for i in a:
            if i in b:
                res.append(i)
        return res